libs/ledger-live-common/src/families/cosmos/js-prepareTransaction.ts

```diff
@@ -10,6 +10,38 @@ import {
 } from "./js-buildTransaction";
 import { getEnv } from "../../env";
 import { log } from "@ledgerhq/logs";
+import { CacheRes, makeLRUCache } from "../../cache";
+
+export const calculateFees: CacheRes<
+  Array<{
+    account: Account;
+    transaction: Transaction;
+  }>,
+  {
+    estimatedFees: BigNumber;
+    estimatedGas: BigNumber;
+  }
+> = makeLRUCache(
+  async ({
+    account,
+    transaction,
+  }): Promise<{
+    estimatedFees: BigNumber;
+    estimatedGas: BigNumber;
+  }> => {
+    return await getEstimatedFees(account as CosmosAccount, transaction);
+  },
+  ({ account, transaction }) =>
```

Comment:
Can you please add back the unit test here : https://github.com/LedgerHQ/ledger-live/pull/3180/files

"should return cached result for the same inputs"

Don't forget the afterEach reset

Reply:
we don't actually need it because we have this one on bridge helper test, that cover the same things that we failed to have

```
          makeTest("ref stability on self transaction", async () => {
            const account = await getSynced();
            await expectStability(account, {
              ...bridge.createTransaction(account),
              amount: new BigNumber(1000),
              recipient: account.freshAddress,
            });
          });
  ```